py-polars/polars/internals/expr/expr.py

```diff
@@ -5373,31 +5374,33 @@ def shuffle(self, seed: int | None = None) -> Expr:
             seed = random.randint(0, 10000)
         return wrap_expr(self._pyexpr.shuffle(seed))
 
+    @deprecated_alias(fraction="frac")
     def sample(
         self,
-        fraction: float = 1.0,
+        frac: float | None = None,
```

Comment:
Shouldn't we add `n` here?

Reply:
If we add `n` here as the first argument, like the other `sample` methods, we break the flow of people using this method without keyword arguments.

For example, if they specify `.sample(0.5, False)`, the method would break because `0.5` becomes the input of `n`, and `False` becomes the input for `frac`, and `with_replacement` remains the default value `True` instead of being set to `False`.

I don't think there is a way to neatly handle this behaviour, so this becomes a breaking change. I add the `n` in my next (breaking) PR.

Reply:
Right.. maybe we can already add the `n` as a keyword only argument? That way we can already use the functionality and we can decide how to reorder the args on a breaking release.

Reply:
Yes, that's definitely possible! Will do that.

Reply:
Done. Behaviour remains slightly different from Series/DataFrame because the default values are dfifferent. So we'll have to keep the FutureWarning until we fix the signature..